doc/user/content/ingest-data/warpstream.md

```diff
@@ -0,0 +1,138 @@
+
+    c. **Generate Credentials**: Post deployment, [create credentials](https://docs.warpstream.com/warpstream/how-to/configure-the-warpstream-agent-for-production/configure-authentication-for-the-warpstream-agent#sasl-authentication) for connecting to your WarpStream cluster.
+
+    d. **Test Connection**: Use the provided credentials to connect to the WarpStream cluster on Fly.io. Test this connection using [the WarpStream CLI](https://docs.warpstream.com/warpstream/install-the-warpstream-agent):
+
+    ```bash
+    warpstream kcmd -type diagnose-connection \
+                    -bootstrap-host $CLUSTER_NAME.fly.dev \
+                    -tls -username ccun_XXXXXXXXXX \
+                    -password ccp_XXXXXXXXXX
+    ```
+
+    Change the `bootstrap-host` to the name of your WarpStream cluster on Fly.io.
+
+    e. **Topic Creation**: Establish the `materialize_click_streams` topic:
+
+    ```bash
+    warpstream kcmd -bootstrap-host $CLUSTER_NAME.fly.dev \
+                    -tls -username ccun_XXXXXXXXX \
+                    -password ccp_XXXXXXXXXX \
+                    -type create-topic \
+                    -topic materialize_click_streams
+    ```
+
+    f. **Produce Sample Records**: Generate and push sample records for testing:
+
+    ```bash
+    warpstream kcmd -bootstrap-host $CLUSTER_NAME.fly.dev \
+                    -tls -username ccun_XXXXXXXXXX \
+                    -password ccp_XXXXXXXXXX \
+                    -type produce \
+                    -topic materialize_click_streams_1 \
+                    --records '{"action": "click", "user_id": "user_0", "page_id": "home"},,{"action": "hover", "user_id": "user_0", "page_id": "home"},,{"action": "scroll", "user_id": "user_0", "page_id": "home"}'
+    ```
+
+    The WarpStream CLI uses `,,` as a delimiter between JSON records.
+
+2. #### Integrate with Materialize
+
+    To integrate WarpStream with Materialize, you need to set up a connection to the WarpStream broker and create a source in Materialize to consume the data.
+
+    Head over to the Materialize console and follow the steps below:
+
+    a. Save WarpStream credentials:
+
+    ```sql
+    CREATE SECRET warpstream_username AS '<username>';
+    CREATE SECRET warpstream_password AS '<password>';
+    ```
+
+    b. Set up a connection to the WarpStream broker:
+
+    ```sql
+    CREATE CONNECTION warpstream_kafka TO KAFKA (
+        BROKER '$CLUSTER_NAME.fly.dev:9092',
+        SASL MECHANISMS = "PLAIN",
+        SASL USERNAME = SECRET warpstream_username,
+        SASL PASSWORD = SECRET warpstream_password
+    );
+    ```
+
+    c. Create a source in Materialize to consume messages:
+
+    ```sql
+    CREATE SOURCE warpstream_click_stream_source
+        FROM KAFKA CONNECTION warpstream_kafka (TOPIC 'materialize_click_streams')
+        FORMAT JSON
+        WITH (SIZE = '3xsmall');
+    ```
+
+    d. To verify the ingestion, you can execute an SQL query in Materialize
```

Comment:
The wording of this one doesn't match the others. Should it be "Verify the ingestion and query the data in Materialize" or something.

Reply:
Ah yes, good catch!